src/zenml/stack/stack.py

```diff
@@ -91,8 +91,6 @@ def __init__(
         self._feature_store = feature_store
         self._model_deployer = model_deployer
 
-        self.validate()
```

Comment:
Is there any reason for removing this early validation? I think right now it prevented users from registering an invalid stack for example on the CLI instead of waiting until they try to run a pipeline

Reply:
@schustmi yes, the problem with validating Stack objects when they are created is that this is called even for CLI commands like `zenml stack get/describe/update`, which will make it impossible for users to review and update their current stacks after updating if they are currently incorrect. Validating the stacks during registration and update is still equally powerful but flexible enough to allow users to correct their current stacks after these new validations come into effect.

Reply:
Makes sense, thanks!